include/ddc/kernels/splines/splines_linear_problem.hpp

```diff
@@ -78,6 +78,18 @@ class SplinesLinearProblem
     {
         return m_size;
     }
+
+    /**
+     * @brief Get the required number of rows of the multi-rhs view passed to solve().
+     *
+     * It is useful because SplinesLinearSolver3x3Blocks requires additional rows for internal operation.
+     *
+     * @return The required number of rows of the multi-rhs view.
+     */
+    virtual std::size_t required_number_of_rhs_rows() const
+    {
+        return m_size;
+    }
```

Comment:
```suggestion
    /**
     * @brief Get the required number of rows of the multi-rhs view passed to solve().
     *
     * Implementations may require a number of rows larger than what `size` returns for optimization purposes.
     *
     * @return The required number of rows of the multi-rhs view. It is guaranteed to be greater or equal to `size`.
     */
    std::size_t required_number_of_rhs_rows() const
    {
        std::size_t const nrows = impl_required_number_of_rhs_rows();
        assert(nrows >= size());
        return nrows;
    }

    // in private
    virtual std::size_t impl_required_number_of_rhs_rows() const
    {
        return m_size;
    }
```

To avoid surprises